src/mpc-hc/CDarkListBox.cpp

```diff
@@ -208,3 +196,20 @@ void CDarkListBox::updateToolTip(CPoint point) {
 void CDarkListBox::OnMouseMove(UINT nFlags, CPoint point) {
     updateToolTip(point);
 }
+
+void CDarkListBox::setIntegralHeight() {
+    CWindowDC dc(this);
+    CFont font;
+    CDarkTheme::getUIFont(font, &dc, CDarkTheme::CDDialogFont);
+    CFont* pOldFont = dc.SelectObject(&font);
+    CRect r(0, 0, 99, 99);
+    CString test = _T("W");
+    dc.DrawText(test, test.GetLength(), &r, DT_LEFT | DT_SINGLELINE | DT_NOPREFIX | DT_CALCRECT);
+    SetItemHeight(0, r.Height());
+
+    dc.SelectObject(pOldFont);
+}

```

Comment:
What does this code do? Leftover test code perhaps?

Reply:
No, actually this is a "fix" I came up with, for an MFC issue.  The CListBox doesn't respond to measureitem requests for LBS_OWNERDRAWFIXED style (even though docs say it does, one time, here: https://docs.microsoft.com/en-us/cpp/mfc/reference/clistbox-class?view=vs-2019#measureitem).  It probably has to do with it being created from the dialog instead of dynamically.

One solution is to use LBS_OWNERDRAWVARIABLE, but that causes weirdness at different DPIs.  Specifically, it makes the listbox resize outside its container.  That may be related to LBS_NOINTEGRALHEIGHT which is on by default for variable height, but actually it would seem that should have the opposite effect.  And the dynamic heights don't actually cause the lisbox to resize.

My solution was to directly set the item height for the font used, rather than wait on requests that never come.  That way, the original height of the listbox is honored while allowing the rows to space appropriately.

CListBox is much less "advanced" in its owner draw capabilities compared to CListCtrl, and this widget could probably be replaced with a listctrl, but for now my goal was to handle all the original widgets as they were.